src/index.js

```diff
@@ -157,16 +160,28 @@ export default function nodeResolve ( options = {} ) {
 				importee,
 				Object.assign( resolveOptions, customResolveOptions )
 			)
-				.catch(() => false)
+				.catch(() => FALSE)
 				.then(resolved => {
-					if (options.browser && packageBrowserField) {
-						if (packageBrowserField[ resolved ]) {
-							resolved = packageBrowserField[ resolved ];
+					if (resolved !== FALSE && options.browser && packageBrowserField) {
+						let l;
+						const hasLink = packageBrowserField.hasOwnProperty(resolved);
+						if (hasLink) {
+							l = packageBrowserField[ resolved ];
+						}
+						browserMapCache[l || resolved] = packageBrowserField;
+						if (hasLink) {
+							resolved = l;
+						}
+						if (resolved === false) {
+							return ES6_BROWSER_EMPTY;
```

Comment:
Maybe the whole section starting with `let l` above could be simplified to

```js
if (packageBrowserField.hasOwnProperty(resolved)) {
	if (!packageBrowserField[resolved]) {
		browserMapCache[resolved] = packageBrowserField;
		return ES6_BROWSER_EMPTY;
	}
	resolved = packageBrowserField[ resolved ];
}
browserMapCache[resolved] = packageBrowserField;
```

This would also get rid of the unhelpfully named `l` and the `hasLink`.

Reply:
thx, great job!